csrc/multidevice/utils.cpp

```diff
@@ -116,27 +116,19 @@ std::unordered_map<IterDomain*, int64_t> mapIterDomainToTensorAxis(
   return id_to_axis;
 }
 
-} // namespace
-
-int64_t getShardedLogicalAxis(
```

Comment:
What you have is totally fine. I'd instead do the following:
```
int64_t getShardedLogicalAxis(const TensorView*, const ParallelType parallel_type, DomainType domain_type) {
  const std::vector<IterDomain*>& domain = the domain according to domain_type;
  std::unordered_map<ParallelType, IterDomain*> parallel_type_to_id =
      mapDeviceAndStreamParallelTypeToId(domain);
  ...same logic...
}
```
This is less change and less function overloading (your current PR has two versions of getShardedLogicalAxis). Function overloading isn't bad -- it just comes with [tradeoffs](https://google.github.io/styleguide/cppguide.html#Function_Overloading) and sometimes makes code harder to read.